File: app/src/main/java/com/ntsoftware/vspc/myvspc/screens/discover/holders/base/DiscoverHolder.java
package com.ntsoftware.vspc.myvspc.screens.discover.holders.base;

import android.content.Context;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public abstract class DiscoverHolder extends RecyclerView.ViewHolder {

    protected View root;

    protected Context context;

    public DiscoverHolder(@NonNull View itemView) {
        super(itemView);
        root = itemView;
        context = itemView.getContext();
    }

    public abstract void bind();
}

File: app/src/main/java/com/ntsoftware/vspc/myvspc/screens/discover/holders/base/DiscoverHolder.kt
package com.ntsoftware.vspc.myvspc.screens.discover.holders.base

import android.content.Context
import android.view.View
import androidx.recyclerview.widget.RecyclerView

open abstract class DiscoverHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    protected val root: View = itemView

    protected val context: Context = itemView.context

    public abstract fun bind()
}